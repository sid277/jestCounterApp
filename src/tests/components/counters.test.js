import React from 'react';
import { shallow } from 'enzyme';
import Counters from '../../components/counters';

describe("Test the Counters file",()=>{

  const data = [
      { id: 1, value: 2 },
      { id: 2, value: 3 },
      { id: 3, value: 1 },
      { id: 4, value: 4 }
    ];

  const data1 = [
    {}
  ];
  
  test('should test Counters component snapshot', () => {
    const wrapper = shallow(<Counters counters={data}/>);
    expect(wrapper).toMatchSnapshot();
    });

  test('should test Counters reset button', () => {
    const wrapper = shallow(<Counters counters={[]}/>);
    const disableProp = wrapper.find('[className="btn btn-success m-2"]');
    expect(disableProp).toBeDisabled();
    });

  test('should test the click of reset button of Counters component',() => {
    let mockFn = jest.fn();
    const wrapper = shallow(<Counters counters={data} onReset={mockFn}/>);
    const disableProp = wrapper.find('[className="btn btn-success m-2"]');
    disableProp.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  test('should test the restart button of Counters component',() => {
    const wrapper = shallow(<Counters counters={data}/>);
    const disableProp = wrapper.find('[className="btn btn-primary m-2"]');
    expect(disableProp).toBeDisabled();
  });

  test('should test the click of restart button of Counters component',() => {
    let mockFn = jest.fn();
    const wrapper = shallow(<Counters counters={data1}  onRestart={mockFn}/>);
    wrapper.find('[className="btn btn-primary m-2"]').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

});

