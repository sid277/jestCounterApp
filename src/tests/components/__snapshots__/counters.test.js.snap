// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Counters file should test Counters component snapshot 1`] = `
<div>
  <button
    className="btn btn-success m-2"
    disabled=""
  >
    <i
      aria-hidden="true"
      className="fa fa-refresh"
    />
  </button>
  <button
    className="btn btn-primary m-2"
    disabled="disabled"
  >
    <i
      aria-hidden="true"
      className="fa fa-recycle"
    />
  </button>
  <Counter
    counter={
      Object {
        "id": 1,
        "value": 2,
      }
    }
    key="1"
  />
  <Counter
    counter={
      Object {
        "id": 2,
        "value": 3,
      }
    }
    key="2"
  />
  <Counter
    counter={
      Object {
        "id": 3,
        "value": 1,
      }
    }
    key="3"
  />
  <Counter
    counter={
      Object {
        "id": 4,
        "value": 4,
      }
    }
    key="4"
  />
</div>
`;
